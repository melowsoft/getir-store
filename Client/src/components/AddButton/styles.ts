import styled from "styled-components";
import { media } from "../../styles/style-utils"

export const Button = styled.div`
width: 100%;
background: #1EA4CE;
border-radius: 2px;
display: flex;
justify-content: center;
align-items: center;
font-family: Open Sans;
font-style: normal;
font-weight: 600;
font-size: 12px;
line-height: 20px;
color: #FFFFFF;
cursor: pointer;
border-color: none;

${
  media.sm`
  
  `
}

`