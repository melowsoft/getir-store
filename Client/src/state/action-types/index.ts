export enum ActionType {
  FETCH_PRODUCTS = 'fetch_products',
  FETCH_PRODUCTS_SUCCESS = 'fetch_products_success',
  FETCH_PRODUCTS_ERROR = 'fetch_products_error',
  ADD_TO_BASKET = 'add_to_basket',
  REMOVE_FROM_BASKET = 'remove_from_basket',
  TOGGLE_BASKET = 'toggle_basket',
  TOGGLE_SIDE_FILTER = 'toggle_side_filter',
}