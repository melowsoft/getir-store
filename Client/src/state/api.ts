

const {
  REACT_APP_API_DEV_BASE_URL,
  REACT_APP_API_PROD_BASE_URL,
  NODE_ENV
} = process.env

export const BASE_URL: any = NODE_ENV === "production"
? REACT_APP_API_PROD_BASE_URL
: REACT_APP_API_DEV_BASE_URL