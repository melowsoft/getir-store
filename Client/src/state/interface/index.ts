export interface Product {
  tags: string[];
  price: number;
  name: string;
  description: string;
  slug: string;
  added: number;
  manufacturer: string;
  itemType: string;
}

export interface BasketPick {
  name: string;
  price: number;
  quantity: number;
}

export interface CartItem {
  name: string;
  price: number;
}