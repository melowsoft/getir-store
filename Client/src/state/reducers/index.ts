import { combineReducers } from 'redux'
import  productsReducer from './productsReducers'
import basketReducer from "./basketReducers"
import showBasketReducer from "./showBasketReducer"
import showSideFilterReducer from "./showSideFilterReducer"

const reducers = combineReducers({
  products: productsReducer,
  basket: basketReducer,
  showBasket: showBasketReducer,
  showSideFilter: showSideFilterReducer
});

export default reducers;

export type RootState = ReturnType<typeof reducers>