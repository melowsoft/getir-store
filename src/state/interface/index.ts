export interface Product {
  tags: string[];
  price: 10.99;
  name: string;
  description: string;
  slug: string;
  added: number;
  manufacturer: string;
  itemType: string;
}

export interface BasketPick {
  name: string;
  price: number;
  quantity: number;
}

export interface CartItem {
  name: string;
  price: number;
}