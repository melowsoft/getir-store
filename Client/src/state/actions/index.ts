import { ActionType } from "../action-types"
import {CartItem, Product} from "../interface"

interface FetchProductsAction {
  type: ActionType.FETCH_PRODUCTS;
}
interface ToggleBasketAction {
  type: ActionType.TOGGLE_BASKET;
}
interface ToggleSideFilterAction {
  type: ActionType.TOGGLE_SIDE_FILTER;
  payload: string;
}
interface FetchProductsSuccessAction {
  type: ActionType.FETCH_PRODUCTS_SUCCESS;
  payload: {products: Product[], pagination: {totalPages: number}, tags: string[], brands: string[]};
}
interface FetchProductsErrorAction {
  type: ActionType.FETCH_PRODUCTS_ERROR;
  payload: string;
}

interface AddProductToBasket {
  type: ActionType.ADD_TO_BASKET;
  payload: CartItem;
}
interface RemoveProductFromBasket {
  type: ActionType.REMOVE_FROM_BASKET;
  payload: CartItem;
}

export type Action =   
| FetchProductsAction 
| FetchProductsSuccessAction 
| FetchProductsErrorAction
| AddProductToBasket
| RemoveProductFromBasket
| ToggleBasketAction
| ToggleSideFilterAction
