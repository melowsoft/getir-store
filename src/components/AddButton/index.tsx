import React from 'react'
import { Product } from "../../state/interface"

import {Button} from "./styles"

interface Props {
  title: string;
  onAdd(product: any): any;
  product: Product
}

export const AddButton: React.FC<Props> = ({title, onAdd, product}: Props) => (
  <Button onClick={() => onAdd(product)}>
    {title}
  </Button>
)