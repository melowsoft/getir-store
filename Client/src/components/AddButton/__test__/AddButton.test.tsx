import React from "react"
import  ReactDOM  from "react-dom"
import { AddButton } from "../../AddButton"
import {Product} from "../../../state/interface"
import { render, cleanup } from '@testing-library/react';
import renderer from "react-test-renderer"


const product:Product = {
  tags: [],
  price: 10,
  name: "test name",
  description: "",
  slug: "test slug",
  added: 0,
  manufacturer: "test brand",
  itemType: "test type"
}

afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<AddButton title="test" onAdd={() => console.log("Added!")} product={product}></AddButton>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it("renders button currectly", () => {
  const {getByTestId} = render(<AddButton title="Add product" onAdd={() => console.log("Added!")} product={product}></AddButton>)
  expect(getByTestId("add-btn")).toHaveTextContent("Add product")
})

it("matches snapshot", () => {
  const tree = renderer.create(<AddButton title="Add product" onAdd={() => console.log("Added!")} product={product}></AddButton>).toJSON()
  expect(tree).toMatchSnapshot()
})