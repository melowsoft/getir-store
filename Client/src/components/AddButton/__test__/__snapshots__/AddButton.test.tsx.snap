// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdnxRM hEBIxC"
  data-testid="add-btn"
  onClick={[Function]}
>
  Add product
</div>
`;
